core/src/c/offheap_jni_JemallocWrapper.cpp

```diff
@@ -0,0 +1,25 @@
+#include "offheap_jni_JemallocWrapper.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+JNIEXPORT jlong JNICALL Java_offheap_jni_JemallocWrapper_malloc_10 (JNIEnv *, jclass, jlong size) {
+    const auto result = je_malloc(static_cast<size_t>(size));
+
+    return result == NULL ? -1 : (jlong) result;
+}
+
+JNIEXPORT jlong JNICALL Java_offheap_jni_JemallocWrapper_realloc_10 (JNIEnv *, jclass, jlong address, jlong newSize) {
+    const auto result = je_realloc((void *) address, static_cast<size_t>(newSize));
```

Comment:
gcc will have this as the max value of uint32_t and uint64_t on 32 and 64 bit platforms respectively. So in 64 bit we will be dealing with long's on both sides [and in fact unsigned long on the C++ side] and on 32 bit, the widening of the [assumed] int into an unsigned long won't cause any problems either. I think we are good in both cases.